g11n-ws/modules/md-service-i18n-l2/src/main/java/com/vmware/i18n/l2/service/pattern/PatternServiceImpl.java

```diff
@@ -207,21 +207,29 @@ private Map<String, Object> getCategories(List<String> categoryList, Map<String,
 		Map<String, Object> resultMap = new LinkedHashMap<>();
 		Map<String, Object> categoriesMap = (Map<String, Object>) patternMap.get(ConstantsKeys.CATEGORIES);
 		Map<String, Object> supplementMap = (Map<String, Object>) categoriesMap.get(ConstantsKeys.SUPPLEMENT);
-		Map<String, Object> suppMap = new HashMap<>();
-		for (String cat : categoryList) {
-			suppMap.put(cat, supplementMap.get(cat));
-		}
+		try {
+			if (categoryList.contains(ConstantsKeys.CURRENCIES) || categoryList.contains(ConstantsKeys.MEASUREMENTS) || categoryList.contains(ConstantsKeys.DATE_FIELDS) && !categoryList.contains(ConstantsKeys.PLURALS)) {
+				categoryList.add(ConstantsKeys.PLURALS);
+			}
+			if (categoryList.contains(ConstantsKeys.PLURALS) && !categoryList.contains(ConstantsKeys.NUMBERS)) {
+				categoryList.add(ConstantsKeys.NUMBERS);
+			}
+		} catch (UnsupportedOperationException e) {//catch the exception when refetch plural and dateFields by language for getPatternWithLanguageAndRegion API
 
-		if (categoryList.contains(ConstantsKeys.CURRENCIES) && !categoryList.contains(ConstantsKeys.NUMBERS)) {
-			categoryList.add(ConstantsKeys.NUMBERS);
-			suppMap.put(ConstantsKeys.NUMBERS, supplementMap.get(ConstantsKeys.NUMBERS));
 		}
 
 		for (String cat : categoryList) {
 			if (!CommonUtil.isEmpty(categoriesMap.get(cat))) {
 				resultMap.put(cat, categoriesMap.get(cat));
 			}
 		}
+
+		Map<String, Object> suppMap = new HashMap<>();
+		for (String cat : categoryList) {
+			if (!CommonUtil.isEmpty(supplementMap.get(cat))) {
```

Comment:
Can you move the if (!CommonUtil.isEmpty(supplementMap.get(cat))) {suppMap.put(cat, supplementMap.get(cat));
			} to line 224